### PR TITLE
Fixes #RHINENG-17375 - Unit and integration tests for Rules Engine

### PR DESCRIPTION
**Pull request by upadhyeammit**
## PR Title :boom:

Please title this PR with a summary of the change, along with the JIRA card number.

Suggested formats: 

1. Fixes/Refs #RHIROS-XXX - Title
2. RHIROS-XXX Title 

Feel free to remove this section from PR description once done.

## Why do we need this change? :thought_balloon:

Please include the __context of this change__ here.

## Documentation update? :memo:

- [ ] Yes
- [ ] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

## Summary by Sourcery

Add comprehensive unit and integration tests for the rules_engine module with supporting fixtures and documentation, and fix resource oversized classification logic for consistent behavior

Bug Fixes:
- Fix oversized resource classification logic by changing evaluation operator from all to any for consistency with undersized logic

Enhancements:
- Include rules_engine_fixtures module in pytest configuration

Documentation:
- Add README documenting test structure, coverage, and scenarios for the rules_engine tests

Tests:
- Add extensive unit tests for rules_engine covering condition functions, utilization calculations, evaluation logic, candidate generation, solution finding, reporting, and utility helpers
- Add integration tests for end-to-end workflows, error handling, performance edge cases, and complete rule execution pipeline
- Add pytest fixtures for reusable mock data and helper classes across tests

### CODE REVIEW

File: tests/fixtures/rules_engine_fixtures.py
Comment (r14chandra) [inline tests/fixtures/rules_engine_fixtures.py:1]: Functions defined in this fixture file are not being used anywhere in the unit test or integration test. The actual tests are not wired to it. Can we delete this file?

**Review by r14chandra — CHANGES_REQUESTED** [review]

**Comment by r14chandra** [comment]
The EC pipeline is failing, and because of that, the build pipeline is not getting started.

File: ros/rules/rules_engine.py [hunk 1]
```diff
@@ -66,7 +66,9 @@ def readable_evalution(ret):
         (ERROR_KEY_UNDERSIZED, any, (RosKeys.CPU_UNDERSIZED,
                                      RosKeys.MEMORY_UNDERSIZED,
                                      RosKeys.IO_UNDERSIZED)),
-        (ERROR_KEY_OVERSIZED, all, (RosKeys.CPU_OVERSIZED,
+        # FIXED: CHANGED FROM 'all' TO 'any' - IF ANY RESOURCE IS OVERSIZED,
```
[the remark this answers is not in the document]
Comment (upadhyeammit) [reply ros/rules/rules_engine.py:69]: We should had discussion, thats why I added the note

File: tests/fixtures/rules_engine_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+
+    return mock_pm
+
+
+@pytest.fixture
+def mock_lscpu():
+    """Mock LsCPU parser data."""
+    mock_lscpu = Mock()
+    mock_lscpu.info = {
+        'CPUs': '2',
```
[the remark this answers is not in the document]
Comment (upadhyeammit) [reply tests/fixtures/rules_engine_fixtures.py:237]: yeah its as per the insights-core

**Review by r14chandra — APPROVED** [review]

File: ros/rules/rules_engine.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (upadhyeammit) [reply ros/rules/rules_engine.py:69]: I am keeping this, will recite this in one of our meetings.
